DistroLauncher/algorithms.h

```diff
@@ -130,3 +130,33 @@ template <typename T, typename... Args> void push_back_many(std::vector<T>& vec,
     vec.reserve(vec.size() + sizeof...(args));
     (vec.push_back(std::forward<Args>(args)), ...);
 }
+
+// Follows the semantics of std::getline(stream&, string&) while implemented for an iterator instead of the stream
+// itself, i.e. reads from [input] iterator until [delim] is found. Although it doesn't go to the string, it's
+// extracted from the underlying stream pointed by the [input] iterator. [str] is also erased before insertion.
+template <class CharT, class Traits, class Allocator>
+std::istreambuf_iterator<CharT, Traits>& getline(std::istreambuf_iterator<CharT, Traits>& input,
+                                                 std::basic_string<CharT, Traits, Allocator>& str, CharT delim = '\n')
```

Comment:
Not convinced with the function signature. In general, iterators are passed by copy, and an updated copy is returned. So `it = getline(it, str)` is the typical form (see std::find, std::partition, std::merge, etc).

Alternatively we could sin a little by passing the iterator by reference and returning the string: `const std::string str = getline(it)`, to help with const-correctness. I would not go for this option because modifying an iterator argument is anti-idiomatic.

Reply:
Oh you're right, I didn't know getline took iterators. Then your implementation is the most idiomatic.